src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -160,6 +159,14 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             // Language Worker Hosted Services need to be intialized before WebJobsScriptHostService
             ScriptHostBuilderExtensions.AddCommonServices(services);
 
+            services.AddSingleton<IFunctionMetadataProvider>(sp =>
+            {
+                return new FunctionMetadataProvider(
+                    ActivatorUtilities.GetServiceOrCreateInstance<ILogger<FunctionMetadataProvider>>(sp),
```

Comment:
`ILogger<T>` is known to be in the container, so you can use:
```suggestion
                    sp.GetRequiredService<ILogger<FunctionMetadataProvider>>(),
```